### ISSUE TITLE
Array is losing items when using flatten+update

### ISSUE THREAD

**Issue by alexandramartinez** [mulesoft-labs/data-weave-rfc#65]
Can be reproduced when using the following script:

```dataweave
%dw 2.0
output json

var limits = flatten([ [ "1", "1" ], [ "2", "2" ] ]) // result: "[1, 1, 2, 2]"
---
{} update {
    case ."$(limits[0])"! -> limits // should be "1": [1, 1, 2, 2] but loses the first '1'
}
```

if you keep changing the index number in `case ."$(limits[0])"!`, it will keep changing the output to have less items the bigger the index.

**Comment by alexandramartinez** [comment]
Pasting here more info from Artur Bielobrov from the community:

---

We think that flatten is reproducing the data stream here with the case expression using the same data stream from two different expressions:
- `limits[0]`
- `limits`

Really strange behavior that could be fixed either by adding `log()` to limits expression or using the limits inside of the update operator one more time.

**Comment by machaval** [comment]
Thanks this indeed was a bug and is going to be fixed on 2.10 version of DW. As a simple workarround just change it to
```dataweave
%dw 2.0
output json

var limits = flatten([ [ "1", "1" ], [ "2", "2" ] ]) // result: "[1, 1, 2, 2]"
var firstLimit = "$(limits[0])"
---
{} update {
    case ."$(firstLimit)"! -> limits // should be "1": [1, 1, 2, 2] but loses the first '1'
}
```
